mephisto/utils/scripts.py

```diff
@@ -19,154 +20,83 @@
-            reqs = db.find_requesters(requester_name=requester_name)
+            reqs = db.find_requesters(provider_type=provider_type)
+            # TODO (#93) proper logging
             if len(reqs) == 0:
                 print(
-                    f"No requesters found under name {requester_name}, "
-                    "have you registered with `mephisto register`?"
+                    f"No requesters found for provider type {provider_type}, please "
+                    f"register one. You can register with `mephisto register {provider_type}`, "
+                    f"or `python mephisto/client/cli.py register {provider_type}` if you haven't "
+                    "installed Mephisto using poetry."
                 )
                 exit(1)
-            provider_type = reqs[0].provider_type
-
-        # provider type and requester name now set, ensure architect
-        if architect_type is None:
-            if provider_type == "mock":
-                architect_type = "local"
-            elif provider_type == "mturk_sandbox":
-                architect_type = "heroku"
-            elif provider_type == "mturk":
-                architect_type = "heroku"
+            elif len(reqs) == 1:
+                req = reqs[0]
+                requester_name = req.requester_name
+                print(
+                    f"Found one `{provider_type}` requester to launch with: {requester_name}"
+                )
             else:
-                architect_type = "local"
-
-            # TODO (#93) proper logging
+                req = reqs[-1]
+                requester_name = req.requester_name
+                print(
+                    f"Found many `{provider_type}` requesters to launch with, "
+                    f"choosing the most recent: {requester_name}"
+                )
+    else:
+        # Ensure provided requester exists
+        reqs = db.find_requesters(requester_name=requester_name)
+        if len(reqs) == 0:
             print(
-                f"No architect specified, defaulting to architect "
-                f"`{architect_type}` for provider `{provider_type}`"
+                f"No requesters found under name {requester_name}, "
+                "have you registered with `mephisto register`?"
             )
+            exit(1)
+        provider_type = reqs[0].provider_type
 
-        if provider_type in ["mturk"]:
-            input(
-                f"This task is going to launch live on {provider_type}, press enter to continue: "
-            )
+    if provider_type in ["mturk"]:
+        input(
+            f"This task is going to launch live on {provider_type}, press enter to continue: "
+        )
+    if provider_type in ["mturk_sandbox", "mturk"] and architect_type != 'heroku':
+        input(
+            f"This task is going to launch live on {provider_type}, but your '
+            f'provided architect is {architect_type}, are you sure you '
+            'want to do this? : "
```

Comment:
Really like this message 👍

Reply:
We used to be able to directly override this after-the-fact, but Hydra has already loaded in the configuration at this point and there's no clean way to mass override _yet_ (waiting on some beta functionality in Hydra)